disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/StorageConfiguration.java

```diff
@@ -8,11 +8,18 @@
 import com.google.auto.value.AutoValue;
 import io.opentelemetry.contrib.disk.buffering.internal.files.DefaultTemporaryFileProvider;
 import io.opentelemetry.contrib.disk.buffering.internal.files.TemporaryFileProvider;
+import java.io.File;
 import java.util.concurrent.TimeUnit;
 
 /** Defines how the storage should be managed. */
 @AutoValue
 public abstract class StorageConfiguration {
+
+  /**
+   * The root storage location for buffered telemetry.
+   */
+  public abstract File getRootDir();
```

Comment:
This is cool! After using this tool a couple of times, I just ended up passing the same root file everywhere which wasn't too nice, this is much cleaner.